### PR TITLE
feat(multiple): switch to angular's new built-in control-flow

### PR DESCRIPTION
**Pull request by mhaertwig**
Reworks the library to use the new built-in control flow instead of `ngIf`, `ngSwitch` and `ngFor`. This has the advantage of not requiring directives to be imported and instantiated and increases the overall performance of the library.

### CODE REVIEW

**Comment by mhaertwig** [comment]
The first two commits in this PR were created using the automatic migration tool for the angular control-flow. The other PRs are manual changes.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "test-firefox": "yarn -s test --firefox",
     "tslint": "tslint -c tslint.json --project ./tsconfig.json",
     "lint": "yarn -s tslint && yarn -s bazel:format-lint",
-    "format:prettier": "prettier --write \"**/*.{js,ts,css,scss,json,md,html}\"",
+    "format:prettier": "prettier --write \"**/*.{js,ts,css,scss,json,md}\" && yarn -s format:prettier:html",
+    "format:prettier:html": "prettier --write \"./src/**/*.html\" --parser angular",
```
Comment (mhaertwig) [inline package.json:31]: We now use prettier's angular parser for html files to correctly format the new control-flow syntax.

File: tools/markdown-to-html/transform-markdown.ts [hunk 2]
```diff
@@ -28,12 +28,12 @@ if (require.main === module) {
...
+    // We use ```angular code blocks to enable correct syntax formatting with prettier. Since markedjs/marked does
+    // not support custom code block languages, we replace all ```angular code blocks with ```html code blocks.
+    inputFileString = inputFileString.replace(/```angular/g, '```html');
     const outputPath = join(bazelBinPath, inputPath.replace(markdownExtension, '.html'));
-    const htmlOutput = markdownRenderer.finalizeOutput(
-      marked(readFileSync(inputPath, 'utf8')),
-      inputPath,
-    );
-
+    const htmlOutput = markdownRenderer.finalizeOutput(marked(inputFileString), inputPath);
```
Comment (mhaertwig) [inline tools/markdown-to-html/transform-markdown.ts:36]: A bit hacky but prettier breaks the formatting of ```html blocks if they use the new control-flow syntax. That's why I replaced these blocks with ```angular and undo this change here.

Comment (jeripeierSBB) [reply tools/markdown-to-html/transform-markdown.ts:36]: but the visual output still looks the same?
